bundles/org.openhab.core.config.core/src/main/java/org/eclipse/smarthome/config/core/Configuration.java

```diff
@@ -127,74 +116,67 @@ public Object remove(Object key) {
     }
 
     public Object remove(String key) {
-        synchronized (this) {
-            return properties.remove(key);
-        }
+        return properties.remove(key);
     }
 
     public Set<String> keySet() {
-        synchronized (this) {
+        synchronized (properties) {
             return Collections.unmodifiableSet(new HashSet<>(properties.keySet()));
         }
     }
 
     public Collection<Object> values() {
-        synchronized (this) {
+        synchronized (properties) {
             return Collections.unmodifiableCollection(new ArrayList<>(properties.values()));
         }
     }
 
     public Map<String, Object> getProperties() {
-        synchronized (this) {
+        synchronized (properties) {
             return Collections.unmodifiableMap(new HashMap<>(properties));
         }
     }
 
-    public void setProperties(Map<String, Object> properties) {
-        for (Entry<String, Object> entrySet : properties.entrySet()) {
-            this.put(entrySet.getKey(), entrySet.getValue());
-        }
-        for (Iterator<String> it = this.properties.keySet().iterator(); it.hasNext();) {
-            String entry = it.next();
-            if (!properties.containsKey(entry)) {
-                it.remove();
-            }
+    public void clear() {
```

Comment:
Do we need an public accessible `clear` method? Why?

Reply:
Since the class is mutable and I needed the method to simplify setProperties(), I saw no need to hide it. But if you think it shouldn't be exposed, I can make it private.

Reply:
As it is used in `setProperties` only, why not calling `properties.clear()` direct?
I would prefer not to add another method if it is used at one place only AND just calls another method (so itself a one liner).
Public would make sense only if we want to provide the option that external users call `clear()` but there has not been any need till now - so we should not just add it because we can.

Reply:
I don't know how many clients do something like setProperties(emptyMap()) which could be improved by having a clear() method, but maybe it's not necessary to add the method in this PR. I removed it.